server/seeder.js

```diff
@@ -1,31 +1,60 @@
-const connection = require('./database.js');
-const mockImages = require('./s3mock.js');
-
-const getRandomImageURL = function(){
-  return mockImages[Math.floor((Math.random() * mockImages.length))];
-}
-
-const getRandomImageCount = function(){
-  return Math.floor(Math.random()*(5-3+1)+3);
-}
-
-const seedDatabase = function(){
-  //for 100 entries.
-  for (let i = 0; i < 100; i++){
-    let imageCount = getRandomImageCount();
-    let imageURLs = [];
-    for (let j = 0; j < imageCount; j++){
-      imageURLs.push(getRandomImageURL());
-    }
-    let query = `insert into images (path) VALUES ('${JSON.stringify(imageURLs)}');`
-    connection.query(query, function(err,results){
-      if(err){
-        console.log(err);
-      } else {
-        console.log(results);
-      }
-    })
-  }
-}
-
-seedDatabase();
+const databaseConnection = require('./database.js');
+const s3mock = require('./s3mock.js');
+
+const getRandomImageURL = function () {
+  return s3mock.mockImages[Math.floor((Math.random() * s3mock.mockImages.length))];
+};
+
+const getRandomImageCount = function () {
+  return Math.floor(Math.random() * (5 - 3 + 1) + 3);
+};
+
+const truncateImageTable = function () {
+  const initializeQuery = 'TRUNCATE TABLE images;';
+  databaseConnection.query(initializeQuery, (err, results) => {
+    if (err) {
+      throw (err);
+    } else {
+      console.log(results);
+    }
+  });
+};
+
+const setTableStart = function (min) {
+  const initializeQuery = `ALTER TABLE images AUTO_INCREMENT = ${min};`;
+  databaseConnection.query(initializeQuery, (err, results) => {
+    if (err) {
+      throw (err);
+    } else {
+      console.log(results);
+    }
+  });
+};
+
+const seedDatabase = function (min = 0, max = 99) {
+  truncateImageTable();
```

Comment:
As this is written, you have no guarantee the table has been truncated before you start seeding the database. This could cause problems. I think it would be better to pass the seeding function into `truncateImageTable` as a callback, so you are sure that it runs only after the table has been truncated.

Reply:
Modified Seeder to perform asynchronous behavior to ensure insure that truncate image table completes before beginning seeding function